scim-client/src/main/java/edu/psu/swe/scim/util/ScimExtensionUtil.java

```diff
@@ -0,0 +1,40 @@
+package edu.psu.swe.scim.util;
+
+import java.util.Collection;
+import java.util.Map;
+
+import edu.psu.swe.scim.spec.extension.ScimExtensionRegistry;
+import edu.psu.swe.scim.spec.resources.ScimExtension;
+import edu.psu.swe.scim.spec.resources.ScimResource;
+
+/**
+ * 
+ * @author mez5001
+ *
+ * This class exposes the SCIM extension methods without requiring that the project using the client knows 
+ * how the ScimExtensionRegistry works
+ */
+public class ScimExtensionUtil {
+
+  /**
+   * Register a single extension for a single resource 
+   * @param rClass ScimResource class
+   * @param eClass Extension class
+   */
+  public static void registerExtension(Class<? extends ScimResource> rClass, Class<? extends ScimExtension> eClass) {
+    ScimExtensionRegistry.getInstance().registerExtension(rClass, eClass);
+  }
+  
+  /**
+   * Register multiple extensions through a map
+   * @param extensions Map containing all of the ScimResources and the accompanying Extensions
+   */
+  public static void registerExtensions(Map<Class<? extends ScimResource>, Collection<Class<? extends ScimExtension>>> extensions) {
```

Comment:
This method signature is a bit awkward ... the user is going to work just as hard (or harder) wrapping up the Map as you are unwrapping it.  Might I suggest that we change the signature of the method on line 24 to:

```
public static void registerExtension(Class<? extends ScimResource> rClass, Class<? extends ScimExtension> eClass
```

And that we eliminate this more complex method.  This would require one line of code for each resource class which I think is reasonable.